### ISSUE TITLE
SESOmitIndex should be ignored unless generateSES is true

### ISSUE THREAD

**Issue by seancorfield** [framework-one/fw1#307]
See Nando Breiter's thread on the mailing list 2/19/2015 where he set `SESOmitIndex` to `true` with `generateSES` set to `false` and got unexpected results in `buildURL()`.

**Comment by seancorfield** [comment]
Nando also noted that "There is one other niggle I could report. I'm fine that it is this way, but baseURL = "/" eliminates index.cfm in the URL even if SESOmitIndex = false. I've tested this by setting baseURL = "", which causes index.cfm to reappear in the url, while changing it back to baseURL = "/" eliminates it again. 

I guess this makes sense because cgi.script_name includes index.cfm. But like I said, I'm fine with that."

**Comment by dnando** [comment]
I'm not sure why the value of SESOmitIndex should be tied to generateSES. Personally, I like being able to omit index.cfm from the url and leave only the query string. It helps, a bit, to mask the application server in use, and eliminates an unnecessary redundancy in the urls, shortening them. In my experience, SESOmitIndex = true has always worked fine alone. That said, it may not work in all scenarios.

**Comment by seancorfield** [comment]
Now that #306 is fixed and 3.0 is released, I'm looking at this ticket again and I'm still leaning toward ignoring `SESOmitIndex` when `generateSES` is `false` since that's how it was supposed to work and the current behavior is a bug as far as I'm concerned.

Given that you have your `baseURL = "/"` workaround @dnando do you have any strong objection to me changing the behavior of `SESOmitIndex` like this? Given your comment about using `SESOmitIndex : true` above, I'm wondering whether leaving things in the *status quo* might be a more generally acceptable resolution for this?

**Comment by seancorfield** [comment]
Somewhat ironically, the `examples/Application.cfc` file sets `SESOmitIndex = true` while `generateSES = false` so there's a precedent for the *status quo* at least...

**Comment by dnando** [comment]
As I think I wrote before, SESOmitIndex = true while generateSES = false allows
one to easily hide index.cfm in the URL without the need to figure out how
to rewrite the URL on the webserver. I like that because of the shorter
url, and because it obscures the application server somewhat. So I'm in
favor of the *status quo.*

I took a stab at rewriting the URL on Nginx on a recent project with
generateSES
= true, and didn't get it to work in my first few attempts. I wanted to
figure it out, but it wasn't worth spending a lot of time doing so. So I
dropped it, and left SESOmitIndex = true and generateSES = false. I'm
managing many applications and domains, so having a relatively simple
configuration that just works is a plus.

Unless someone has a practical objection to how it works now, I'd say leave
it as it is.




Aria Media Sagl
Via Rompada 40
6987 Caslano
Switzerland

+41 (0)91 600 9601
+41 (0)76 303 4477 cell
skype: ariamedia
